[EI-74,EI-83] Add Mermaid architectural diagram

jazzsequence opened the pull request:
This adds a [Mermaid diagram](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/) to represent the architecture for the Drupal SDK similar to https://github.com/pantheon-systems/edge-integrations-wordpress-sdk/pull/15

~Looking in particular for feedback on the module descriptions to make sure that it's accurate. Also, any feedback you have about the diagram itself. I'm trying to represent the idea that Smart Content is used for the Smart Content CDN module but does not have a relationship to EI or the global library (and also that the Smart Content Preview module extends Smart Content, and _can be used_ for Smart Content CDN, but is not a direct requirement), making this diagram a bit trickier than the WP one.~

This PR has been updated with the following changes:
* Includes #5
* Removes references in architecture diagram to Smart Content
* Updates Smart Content CDN in diagram to an _optional_ package, rather than _required_
* Updates documentation to describe Smart Content CDN as an _optional_ implementation
* Removes Smart Content Preview from architecture diagram (and comments out description)
* Updates setup instructions to remove references to Smart Content CDN as a _bundled_ package
* Adds documentation for requiring the libraries independent of the SDK

[Preview](https://github.com/pantheon-systems/edge-integrations-drupal-sdk/blob/mermaid-architectural-diagram/README.md)

jazzsequence committed:
add initial mermaid diagram 

...and copy pasta description from wp sdk

jazzsequence committed:
update descriptions

jazzsequence committed:
use default badge style

jazzsequence committed:
Merge branch 'main' into mermaid-architectural-diagram
resolve merge conflict in readme

jazzsequence committed:
lower the priority of the path repository

jazzsequence committed:
require pantheon-edge-integrations, not smart_content_cdn
suggest smart_content_cdn as a way to implement.

jazzsequence committed:
Merge branch 'ei-83-decouple-smart_content_cdn-as-dep' into mermaid-architectural-diagram

jazzsequence committed:
updates mermaid diagram
* removes drupal
* removes smart_content
* updates smart_content_cdn as an optional dependency
* removes smart_content_preview

jazzsequence committed:
update descriptions of components
* removes Smart Content
* updates Smart Content CDN description and links to Smart Content project

jazzsequence committed:
comment out the Smart Content Preview documentation

jazzsequence committed:
remove no-longer-relevant dev note

RobLoach reviewed (approved):
Thanks for taking this one on :+1:

jazzsequence committed:
update setup docs
we don't include smart_content_cdn via #3, so remove that from the description

jazzsequence committed:
add note that you can install all of these things separately

jazzsequence commented:
Alrighty, I've finished tweaking the readme. The major changes were made prior to @RobLoach's 👍 and I've detailed them in the description. The only additional details I added after the ✅ were updating the setup docs to not reference that Smart Content CDN is bundled and added a note that you can install everything independent of the SDK.

@reinette Tagging you in for a quick 👍 to make sure this satisfies all of your goals. In the future, if/when `pantheon-systems/pantheon_drupal_edge_integrations` is added to the SDK, the diagram would be modified to look like this:

```mermaid
flowchart TB
agcdn[/Pantheon Advanced Global CDN\]-->ei[Pantheon Edge Integrations global library]
ei-->drupal_ei(Pantheon Drupal Edge Integrations)
ei-. optional .->smart_content_cdn(Smart Content CDN)
```

...and if/when Smart Content Preview is released, it might look something like this:

```mermaid
flowchart TB
agcdn[/Pantheon Advanced Global CDN\]-->ei[Pantheon Edge Integrations global library]
ei-->drupal_ei(Pantheon Drupal Edge Integrations)
ei-. optional .->smart_content(Smart Content)
smart_content-. optional .->smart_content_cdn(Smart Content CDN)
smart_content-. optional .->smart_content_preview(Smart Content Preview)
```

jazzsequence commented:
(There's also a way in Mermaid to break out sections of a diagram and isolate them as a group, so it might also be that we do something like that for the entire Smart Content ecosystem to show that it can be used collectively but independent of the core functionality. https://mermaid-js.github.io/mermaid/#/flowchart?id=subgraphs)
